### PR TITLE
Remove default props

### PR DESCRIPTION
**Pull request by slightlytyler**
Removes default props if their value is the default. This should match JSX that was input (or at least it does in my scenario).

### CODE REVIEW

File: index-test.js [hunk 1]
```diff
@@ -449,6 +453,12 @@ describe(`reactElementToJSXString(ReactElement)`, () => {
     ).toEqual(`<TESTCOMPONENT />`);
   });
 
+  it('reactElementToJSXString(<DefaultPropsComponen />)', () => {
+    expect(
+      reactElementToJSXString(<DefaultPropsComponent />)
```
Comment (vvo) [inline index-test.js]: > Removes default props if their value is the default.

I am not sure to follow what you are trying to prove in this test, shouldn't we use test='test' somewhere?

**Comment by vvo** [comment]
Thanks for your contribution! I am looking forward at understanding the usecase.

File: index-test.js [hunk 2]
```diff
@@ -449,6 +453,12 @@ describe(`reactElementToJSXString(ReactElement)`, () => {
...
+    ).toEqual(`<DefaultPropsComponent />`);
```
Comment (vvo) [inline index-test.js]: Shouldn't the output be DefaultPropsComponent test=test here?

Comment (slightlytyler) [inline index-test.js]: The point is that if a passed property is equivalent to the default it should not be shown. Maybe it should be an option than the default behavior. `<DefaultPropsComponent />` and `<DefaultPropsComponent test="test" />` are equivalent but the latter is showing props that are inferred from the defaults.

**Comment by slightlytyler** [comment]
The use case for me is in my documentation components. I have an `Example` component that renders the children as well as renders their code in a block.
```javascript
<Example>
  <Button>Button</Button>
</Example>
```
screen shot 2016-06-27 at 9 39 52 am
Without my changes you see all the default props, which were not passed in but inferred, in the code block.

**Comment by vvo** [comment]
Ok, can you create an option like "showDefaultProps", should be true by default if that was the current behavior.

**Comment by slightlytyler** [comment]
@vvo should be good to go now

**Comment by slightlytyler** [comment]
oops need to update docs actually

**Comment by vvo** [comment]
awesome, I love the fact that you were able to hack this new option in the current code without my help :) GG!

**Comment by vvo** [comment]
Where is the documentation you are working on?

**Comment by slightlytyler** [comment]
@vvo it's easy to augment good code :wink: did you see the documentation? I added a blurb to the options section

**Comment by vvo** [comment]
yes, perfect :)

**Pull request merged**